### ISSUE TITLE
Auto-Grading Fails to finish for large classes.

### ISSUE THREAD

**Issue by irunestone** [RunestoneInteractive/RunestoneServer#1163]
Error reported in course ComputationalThinking on page index by user profsethi profsethi@sethi.org
Hello,
Thank you for providing this great service. I'm using it in three of my classes and the auto-grading has failed on it. Individual grading works but individually grading 60 students is a chore. The autograder seems to be running but the dialog box doesn't come up and it doesn't actually grade any of the assignments via the autograder.
It would be *wonderful* if this could be fixed and I'd be happy to help fix it,
Rick.

**Comment by psmaker** [comment]
I have over 100 students in my class. I have the same issue. 

I have a couple of workarounds for this. 

**Method 1**: I select 1 question (but do NOT select a student) to be graded at a time. This grades all of my students for this 1 question. Then I select the next question and repeat. This is not too bad.

![voila_capture 2018-11-22_08-39-57_am](https://user-images.githubusercontent.com/28868409/48915069-3a60d480-ee32-11e8-8bdf-7cdc695c3522.png)

**Method 2**: Last year, when I ran into this problem, I posed it to one my student clerks who is amazing. Within about a day, he wrote for a little javascript that I paste into the java console when I'm on the Grading page. I first select the assignment I want to grade (but do not select the question or student). I open the java console on the browser. I paste the script into and have it run. It will then grade each student one by one. I've attached the javascript below.

![voila_capture 2018-11-22_08-52-36_am](https://user-images.githubusercontent.com/28868409/48915591-11414380-ee34-11e8-9b2a-9838f2588c3b.png)


```
(function() {
  const studentSelect = document.getElementById("gradingcolumn3")
  const studentOptions = studentSelect.options

  const origAlert = window.alert

  const gradeByStudentsIndex = (idx) => {
    if (idx < studentOptions.length) {
      studentSelect.selectedIndex = idx;
      gradeIndividualItem();

      window.alert = (msg) => {
        window.alert = origAlert;
        console.log(studentOptions[idx].innerHTML + " (" + (idx + 1) + "/" + (studentOptions.length) + ")" + " - " + msg)
        gradeByStudentsIndex(idx + 1)
      }

      autoGrade();
    }
  }

  gradeByStudentsIndex(0);
})()
```

**Comment by bnmnetp** [comment]
Thanks for inspiration.   I have implemented #1173 so that when an assignment is selected and nothing else we will iterate over the students and grade the assignment student by student automatically.  This isn't the ultimate fix for this but its a decent workaround for now.
